application/docs/methodology/best-practices/kubernetes-probes/real-world-examples/background-job-processor.md

```diff
@@ -0,0 +1,379 @@
+
+def callback(ch, method, properties, body):
+    global last_processed_time
+    try:
+        # Process message
+        process_job(body)
+        jobs_processed.inc()
+        last_processed_time = time.time()
+        ch.basic_ack(delivery_tag=method.delivery_tag)
+    except Exception as e:
+        jobs_failed.inc()
+        print(f"Job failed: {e}")
+        ch.basic_nack(delivery_tag=method.delivery_tag, requeue=True)
+
+@app.route('/metrics')
+def metrics():
+    return generate_latest()
+
+@app.route('/ready')
+def readiness():
+    if rabbitmq_connected:
+        queue_connected.set(1)
+        return "READY", 200
+    else:
+        queue_connected.set(0)
+        return "RabbitMQ not connected", 503
+```
+
+## Testing
+
+### Local Testing
+
+```bash
+# Start RabbitMQ
+docker run -d -p 5672:5672 rabbitmq:3
+
+# Set environment variables
+export RABBITMQ_HOST=localhost
+
+# Start the worker
+python worker.py
+
+# In another terminal, test health endpoints
+curl http://localhost:8080/healthz
+curl http://localhost:8080/ready
+
+# Publish a test message
+python -c "
+import pika
+connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
+channel = connection.channel()
+channel.queue_declare(queue='jobs')
+channel.basic_publish(exchange='', routing_key='jobs', body='test job')
+connection.close()
+"
+
+# Stop RabbitMQ and check readiness
+docker stop <container-id>
+curl http://localhost:8080/ready  # Should return 503
+```
+
+### Kubernetes Testing
+
+```bash
+# Deploy the worker
+kubectl apply -f deployment.yaml
+
+# Check worker logs
+kubectl logs -f <pod-name>
+
+# Publish messages via RabbitMQ service
+kubectl exec -it <rabbitmq-pod> -- rabbitmqadmin publish routing_key=jobs payload="test job"
+
+# Monitor probe status
+kubectl describe pod <pod-name> | grep -A 10 "Liveness\|Readiness"
+
+# Check if worker is processing
+kubectl logs <pod-name> | grep "Processing"
+```
+
+## Troubleshooting
+
+### Worker Not Processing Messages
+
+**Symptom**: Pod is running but not processing messages from queue
+
+**Likely Causes**:
+
+1. Queue doesn't exist or wrong queue name
+2. RabbitMQ connection credentials incorrect
+3. Worker thread crashed but Flask server still running
+
+**Solutions**:
+
+- Check worker logs: `kubectl logs <pod-name>`
+- Verify queue exists: `kubectl exec <rabbitmq-pod> -- rabbitmqctl list_queues`
+- Check connection in readiness endpoint: `kubectl exec <pod-name> -- curl localhost:8080/ready`
+
+### Frequent Restarts Due to Liveness
+
+**Symptom**: Pods restart frequently even though RabbitMQ is healthy
+
+**Likely Causes**:
+
+1. `last_processed_time` threshold too aggressive
+2. Queue is empty (no messages to process)
+3. Jobs take longer than 5 minutes
+
+**Solutions**:
+
+- Adjust the 5-minute threshold to match your longest job
+- Consider removing time-based liveness check if queue can be idle
+- Add separate "heartbeat" mechanism independent of job processing
+
+### Memory Leaks
```

Comment:
i'm not a big fan of this one
- If 90%+ of the memory is consumed, it will restart the pod, but that was also the initial symptom. So the problem will stay the same, just 10% earlier...
- it breaks a bit separation of concern; IMHO, memory restarts must be detected by our observability stack, not the app itself
- It adds confusion to the nice example above

I'm in favor of removing this example totally.